config/tsup.config.cjs

```diff
@@ -0,0 +1,43 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { defineConfig } from 'tsup';
+
+const cwd = process.cwd();
+
+const INPUT_FILE_PATH = path.join(cwd, 'src/index.ts');
+const INPUT_FILE = fs.existsSync(INPUT_FILE_PATH)
+  ? INPUT_FILE_PATH
+  : path.join(cwd, 'src/index.tsx');
+
+// export default defineConfig({
+//   entry: [INPUT_FILE],
+//   silent: true,
+//   format: ['cjs', 'esm'],
+//   outExtension: (ctx) => {
+//     return { js: `.${ctx.format === 'esm' ? 'es' : ctx.format}.js` };
+//   },
+//   outDir: 'dist',
+// });
+
+const { dependencies = {}, peerDependencies = {} } = JSON.parse(
+  fs.readFileSync(path.join(cwd, 'package.json'), 'utf8')
+);
+
+export default defineConfig({
+  clean: true,
+  outDir: 'dist',
+  sourcemap: true,
+  format: ['esm', 'cjs'],
+  entry: [INPUT_FILE],
+  external: [
+    ...Object.keys(dependencies),
+    ...Object.keys(peerDependencies),
+    'react-textarea-autosize',
+  ],
+  outExtension: (ctx) => {
+    return { js: `.${ctx.format === 'esm' ? 'es' : ctx.format}.js` };
```

Comment:
this is incorrect, if you match the rollup output, you'd want to change this to:

js:  ctx.format === 'esm' ? '.es.js' : '.js'

Reply:
In roll-up config file output extension format is `es.js` for `esm` and `cjs.js` for cjs

Reply:
But for the build of Plate, if run rollup, you'd see that cjs.js is not what you get. It's either es.js or .js.

Reply:
yes I have checked it here in package.json of package

`
  "main": "dist/index.js",
  "module": "dist/index.es.js",
`

Reply:
✨